lib/YaST/Firstboot/LanguageAndKeyboardPage.pm

```diff
@@ -0,0 +1,52 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The class introduces all accessing methods for
+# firstboot Language and Keyboard client
+
+package YaST::Firstboot::LanguageAndKeyboardPage;
+use parent 'YaST::Firstboot::GenericPage';
+use strict;
+use warnings;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init();
+}
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
+    $self->{cb_keyboard} = $self->{app}->combobox({id => 'keyboard'});
+    $self->{cb_language} = $self->{app}->combobox({id => 'language'});
+    $self->{lb_main}     = $self->{app}->label({debug_label => 'Language and Keyboard Layout'});
+    return $self;
+}
+
+sub is_shown {
+    my ($self) = @_;
+    return $self->{lb_main}->exist();
+}
+
+sub check_selected_language {
+    my ($self, $expected_value) = @_;
+    my $selected_language = $self->{cb_language}->value();
+    die "The selected language \"$selected_language\" is not the expected one \"$expected_value\"" unless $expected_value eq $selected_language;
```

Comment:
this is what we were talking today offline for another task, that this functionality should not be on the page but in the test module. So we should collect the data in the page or the controller and in the test module we do the comparison using test assert library and taking into account that we need to see all the errors not die on the first one. We should think about a generic solution here (Please also sync with @JRivrain)

Reply:
We kind of figured it out, but I really think we should organize better when it comes to the scopes of each tickets. I don't think when someone in the team should have a good idea one day that it's not yet implemented anywhere,  we should have to apply to all the tickets in progress.

Reply:
Collect the data from the page might be out of the scope of the ticket (but we will need to do it anyway and unify in future), but to avoid making the assertion on the page was not as it breaks the idea of the layers of the framework. Probably was not clear this point. For our internal working, better discuss in offline than in github IMO.

Reply:
On my side I used part of what is done here, produce a hash in the controller then use it to compare against test data. But I chose to do it with a loop, the usage of is_deeply looked promising but does not allow to show failure directly in Openqa (only in the logs). I thought it might be possible to compare hashes simply like we can do it in Ruby, but it looks like Perl does not allow it simply without some external libs that we don't have (or do we ?). See how I implemented the data comparison here http://waaa-amazing.suse.cz/tests/15255#step/yast2_bootloader2/12, and comment if you want in the draft I posted here https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12435 (but remember, it's a draft :) ).
I think we are using the same approach (create a hash from the collected data and compare in the test module) and the research you made here is valuable, but maybe overkill for this current task where you just need to compare a couple of values.